File: internal/domain/group.go
package domain

import "time"

type Group struct {
	ID             string    `json:"id" db:"id"`
	Name           string    `json:"name" db:"name"`
	Description    string    `json:"description" db:"description"`
	ReceiversCount int       `json:"receivers_count" db:"receivers_count"`
	CreatedAt      time.Time `json:"created_at" db:"created_at"`
	UpdatedAt      time.Time `json:"updated_at" db:"updated_at"`
}

File: webui/src/types/group.ts
import type { Receiver } from './receiver'

export interface Group {
  id: string
  name: string
  description: string
  receivers_count: number
  created_at: string
  updated_at: string
}

export interface GetGroupResponse {
  group: Group
}

export interface ListGroupsResponse {
  groups: Group[] | null
}

export interface CreateGroupRequest {
  name: string
  description: string
}

export interface CreateGroupResponse {
  id: string
}

export interface UpdateGroupRequest {
  name?: string
  description?: string
}

export interface UpdateGroupResponse {
  id: string
}

export interface ListGroupReceiversResponse {
  receivers: Receiver[] | null
}
